File: 04-estructuras-datos/01-fundamentos-y-complejidad/java/ejemplos/CostosBasicos.java
import java.util.ArrayList;
import java.util.List;

public class CostosBasicos {
    public static void main(String[] args) {
        List<Integer> valores = new ArrayList<>(List.of(10, 20, 30, 40, 50));

        // get accede directamente a una posición de ArrayList: costo aproximado O(1).
        System.out.println("Posición 3: " + valores.get(3));

        // contains puede revisar varios elementos hasta encontrar el valor: O(n) en el peor caso.
        System.out.println("Contiene 40: " + valores.contains(40));

        // Insertar al inicio desplaza los elementos existentes: O(n).
        valores.add(0, 5);
        System.out.println(valores);
    }
}

File: 04-estructuras-datos/01-fundamentos-y-complejidad/scala/ejemplos/CostosBasicos.scala
object CostosBasicos:
  def main(args: Array[String]): Unit =
    val valores = Vector(10, 20, 30, 40, 50)

    // Vector permite acceso indexado eficiente sin recorrer explícitamente la colección.
    println(s"Posición 3: ${valores(3)}")

    // contains puede revisar elementos hasta localizar el valor.
    println(s"Contiene 40: ${valores.contains(40)}")
